boards/nxp/frdm_mcxn236/board.c

```diff
@@ -129,8 +129,9 @@ static int frdm_mcxn236_init(void)
 #endif
 
 #if DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(flexcomm3))
+	/* SPI loopback test requires at lease 2*16M. */
```

Comment:
The reason that we set the spi loopback test to 16M is (or at least should be) because that is what the datasheet says this LPSPI can achieve.  So I would rather put an explanation along those lines than to say it is for a specific test.  I.e., we want to set the default board configuration to have the SPI which is broken out to headers on the board to be able to reach the datasheet specified frequency.  And same for the comments in the other files as well.  The commit message should probably keep the explanation still also about the spi loopback test since that's your motivation for the change.